pennylane/ops/qubit/parametric_ops.py

```diff
@@ -829,7 +829,13 @@ def _matrix(cls, *params):
 
         c = qml.math.cos(theta / 2)
         s = qml.math.sin(theta / 2)
-        z = qml.math.zeros([4], like=interface)
+
+        if interface == "torch":
+            # Use convert_like to ensure that the tensor is put on the correct
+            # Torch device
+            z = qml.math.convert_like(qml.math.zeros([4]), theta)
+        else:
+            z = qml.math.zeros([4], like=interface)
```

Comment:
@antalszava for code simplicity, you could replace this with just?
```suggestion
        z = qml.math.convert_like(qml.math.zeros([4]), theta)
```

Reply:
but I guess this introduces overhead.

Reply:
The point of `qml.math` is so that places like here don't have to worry about interfaces.

Especially as this occurs in multiple different operations, maybe we should extract this functionality into `qml.math`. 

I think this is okay to keep for now as a bug fix, but in the future, we should try to extract interface-specific behaviour away from places like this.

Reply:
Having `z = qml.math.convert_like(qml.math.zeros([4]), theta)` raises an error for TensorFlow, see the run attached to the commit right before the commit with branching off was added:
1. https://github.com/PennyLaneAI/pennylane/pull/1927/commits/52c8f9c896e52f4968ce4096a6fe10259c013183 (commit before)
2. https://github.com/PennyLaneAI/pennylane/pull/1927/commits/53b41114a28a48c6093e611e92f7845fcd2b456e (commit after)